### ISSUE TITLE
Unable to acquire lock

### ISSUE THREAD

**Issue by krg7880** [mike-marcacci/node-redlock#38]
**Redis Client**
[ioredis](https://github.com/luin/ioredis)

**Sample Code**
```
const test = async () => {
  const Redis = require('ioredis')
  const Redlock = require('redlock')
  const client = new Redis('localhost:6379')
  const redLock = new Redlock([client])
  try {
    await client.setex('foo', 6000, JSON.stringify({ name: 'Bob' }))
    const lock = await redLock.lock('foo', 2000)
  } catch (e) {
    console.log(e)
  }
}

test()
```
**Error**
```
LockError: Exceeded 10 attempts to lock the resource "foo"
```

I dug into the code to inspect what was going on and it looks like the [call](https://github.com/mike-marcacci/node-redlock/blob/master/redlock.js#L289) to acquire the lock never returns a response (ie: `OK`) although there were no errors. This means the check to increment [votes](https://github.com/mike-marcacci/node-redlock/blob/master/redlock.js#L317) fails and we never get the lock. Any thoughts on this or am I way off?

**Comment by mike-marcacci** [comment]
Hey @krg7880, this would be expected, since you’re trying to use the same redis key as a data store and as a lock. Try `await redLock.lock('locks:foo', 2000)` and let me know if that fixes your problem.

**Comment by mike-marcacci** [comment]
To follow up with a more thorough response, there are a few things it looks like you're trying to do here that aren't exactly standard modus operandi.

As mentioned above, a redis key really can't be used by redlock *and* something else. The purpose of this tool is far broader than locking something that is *stored* in redis (rather, it *uses* redis to gain distributed consensus about any kind of exclusivity). Accordingly, if you're storing something in redis and trying to guarantee exclusive access to it, you'll need two different keys, such as `storage:foo` and `locks:foo`.

It may be that you're aware of this and were just trying to test redlock manually by setting a value in the key you're telling it to use. If this is the case, than there are a couple notes I want to make:

1. [`SETEX`](https://redis.io/commands/setex) takes seconds, not milliseconds, so unless you were trying to run a 10-minute test, you're probably looking for [`PSETEX`](https://redis.io/commands/psetex) 😉
2. By default, redlock used a `retryDelay` of 200ms and a `retryCount` of 10 with a `retryJitter` of 200ms. This means that on average we will try to get a lock for 2 seconds before failing. You can [easily change this](https://github.com/mike-marcacci/node-redlock#configuration) behavior when instantiating redlock. Given this, though, it's completely correct that you would receive that error, since the key is taken for 10 minutes as written, and would still be correct had you used `PSETEX` and held it open for 6 seconds.

Please let me know if this helps or if I'm missing something here!

Cheers,
Mike

**Comment by krg7880** [comment]
@mike-marcacci Thanks for the prompt and detailed follow-up. Yeah, the issue was not providing a unique prefix for the lock key. I was able to get it working this morning by doing just that. With regards to the value of the `TTL` and the Redis commands being utilized (ie: `SETEX`), the values were dummy values I entered while drafting this ticket. At this point, everything seems to be working as expected. Thanks again!
